developer-extension/src/panel/components/tabs/eventsTab/computeFacetState.ts

```diff
@@ -0,0 +1,59 @@
+import type { FacetRegistry, FacetValuesFilter } from '../../../hooks/useEvents'
+import type { Facet, FacetValue } from '../../../facets.constants'
+type SelectionState = 'selected' | 'unselected' | 'partial-selected'
+
+export function computeSelectionState(
```

Comment:
Maybe
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Maybe add a comment about the limitation of computing the `partial-selected`: It only works for direct parent.